vision/src/autogluon/vision/predictor/predictor.py

```diff
@@ -697,7 +698,11 @@ def load(cls, path, verbosity=2):
         if os.path.isdir(path):
             path = os.path.join(path, 'image_predictor.ag')
         with open(path, 'rb') as fid:
-            obj = pickle.load(fid)
+            gpu_count = get_gpu_count_all()
+            if gpu_count > 0:
```

Comment:
General question: Should we even be pickling an object with a torch cuda tensor in the first place? What alternative serialization methods are there?

Reply:
This is actually an awesome question and I had a similar concern as well. Currently, ImagePredictor is saving a single pickle file. This behavior differs from what tabular and text are doing: saving to a directory. Ideally, we should save the estimator with torch.save() method, which saves to a file, along with our ImagePredictor pickle file into a directory. However, this is an api breaking change